lib/ruby_indexer/test/instance_variables_test.rb

```diff
@@ -172,5 +172,20 @@ def test_top_level_instance_variables
       entry = T.must(@index["@a"]&.first)
       assert_nil(entry.owner)
     end
+
+    def test_class_instance_variables_inside_self_method
+      index(<<~RUBY)
+        class Foo
+          def self.bar
```

Comment:
We do check that when entering the singleton context. I added a test to demonstrate the current behaviour (which is to associate the instance variable with the owner owner since we can't process dynamic `def something.bar` declarations.